src/index.ts

```diff
@@ -254,34 +254,26 @@ function getConvexPolyhedronParameters (object: Object3D): ShapeParameters<Shape
 		);
 	}
 
-	// Compute the 3D convex hull.
-	const hull = new ConvexHull().setFromObject(new Mesh(geometry));
-	const hullFaces = hull.faces;
-	const vertices: number[] = [];
-	const faces: number[][] = [];
-
-	let currentFaceVertex = 0;
-	for (let i = 0; i < hullFaces.length; i++) {
-		const hullFace = hullFaces[ i ];
-		const face: number[] = [];
-		faces.push(face);
-
-		let edge = hullFace.edge;
-		do {
-			const point = edge.head().point;
-			vertices.push(point.x, point.y, point.z);
-			face.push(currentFaceVertex);
-			currentFaceVertex++;
-			edge = edge.next;
-		} while ( edge !== hullFace.edge );
+	// Get geometry points
+	const points: Vector3[] = []; 
+	for (let i = 0; i < geometry.attributes.position.array.length; i += 3) {
```

Comment:
A few things to think about here — 

1. The BufferAttribute might be interleaved, so it's safer to use `attribute.count`, `attribute.getX( i )`, etc. instead of accessing `attribute.array` directly. These methods [may also handle normalized attributes in the future](https://github.com/mrdoob/three.js/pull/22874).
2. May need to bake `node.matrixWorld` into the vertices

The code from `setFromObject` did handle those cases, I wonder if we want to keep that method and do something like:

```js
const faces = new ConvexHull().setFromObject(object).collectFaces();
```

Reply:
Thanks for the info! I will revert back to `setFromObject` then yep. Will update the PR soon.